blaze/expr/reductions.py

```diff
@@ -14,6 +17,51 @@
 from .expressions import dshape_method_list, method_properties
 
 
+class Window(Expr):
+    __slots__ = (
+        '_hash',
+        '_child', '_group_by', '_sort_by', 'preceding', 'following',
+        '_asdshape'
+    )
+
+    __inputs__ = '_child', '_group_by', '_sort_by'
+
+    def __str__(self):
+        slots = '_group_by', '_sort_by', 'preceding', 'following'
+        return '%s(%s, %s)' % (
+            type(self).__name__,
+            self._child,
+            ', '.join('%s=%r' % (slot, getattr(self, slot)) for slot in slots)
+        )
+
+    def _dshape(self):
+        return self._asdshape
+
+    @property
+    def _name(self):
+        return self._child._name
+
+    def sort_by(self, key):
```

Comment:
We could also change everything to sort_by, though this doesn't look very good on single columns of things